catroid/src/main/java/org/catrobat/catroid/utils/TextSizeUtil.java

```diff
@@ -69,6 +80,38 @@ public static void enlargeViewGroup(ViewGroup viewGroup) {
 		}
 	}
 
+	public static void enlargeTableLayoutButtonText(TableLayout tableLayout) {
+		if (modifier == 1.0f) {
+			return;
+		}
+
+		Context context = CatroidApplication.getAppContext();
+		WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+		Display display = wm.getDefaultDisplay();
+		DisplayMetrics metrics = new DisplayMetrics();
+		display.getMetrics(metrics);
+
+		if ((metrics.widthPixels / metrics.xdpi) > widthForIncreasedTextSizeInInches) {
+			modifier += modifierIncreaseDelta;
```

Comment:
as modifier is a member variable i see potential issues by modifying it here.
this affects all other methods in the class